demo/02-customer-infra.sh

```diff
@@ -18,3 +18,29 @@ az network vnet create \
   --subnet-name ${CUSTOMER_VNET_SUBNET1} \
   --subnet-prefixes 10.0.0.0/24 \
   --nsg "${NSG_ID}"  --location ${LOCATION}
+
+az keyvault create \
+  --name "${CUSTOMER_KV_NAME}" \
+  --resource-group "${CUSTOMER_RG_NAME}" \
+  --location "${LOCATION}" \
+  --enable-rbac-authorization true
+
+# Get current user's object ID
+CURRENT_USER_ID=$(az ad signed-in-user show --query id -o tsv)
+
+# Assign Key Vault Crypto Officer role to current user
+az role assignment create \
+  --assignee "${CURRENT_USER_ID}" \
+  --role "Key Vault Crypto Officer" \
+  --scope "/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${CUSTOMER_RG_NAME}/providers/Microsoft.KeyVault/vaults/${CUSTOMER_KV_NAME}"
+
+# Wait for role assignment to propagate
+echo "Waiting for RBAC role assignment to propagate..."
+sleep 30
+
+# Create RSA key in Key Vault
+az keyvault key create \
+  --vault-name "${CUSTOMER_KV_NAME}" \
+  --name "encryption-key" \
```

Comment:
This name is too generic.

Let's name it something like:
etcd-data-kms-encryption-key